torchmultimodal/models/video_vqvae.py

```diff
@@ -4,10 +4,161 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
+from typing import Any, Dict, List, Tuple, Union
+
 from torch import nn, Tensor
 from torchmultimodal.modules.layers.attention import AxialAttentionBlock
 
-from torchmultimodal.modules.layers.conv import SamePadConv3d
+from torchmultimodal.modules.layers.conv import SamePadConv3d, SamePadConvTranspose3d
+
+
+class VideoEncoder(nn.Module):
+    """Encoder for Video VQVAE. Stacks specified number of ``SamePadConv3d`` layers
+    followed by a stack of ``AttentionResidualBlocks`` and a final ``SamePadConv3d``
+    layer before the codebook. The residual blocks use Axial Attention to enhance
+    representations of video data without significantly increasing computational
+    cost. Follows VideoGPT's implementation:
+    https://github.com/wilson1yan/VideoGPT/blob/master/videogpt/vqvae.py
+
+    Args:
+        in_channels (Tuple[int]): tuple of input channel dimension for each conv layer
+        out_channels (Tuple[int]): tuple of output channel dimension for each conv layer
+        kernel_sizes (Tuple[int or Tuple[int]]): tuple of kernel sizes for each conv layer
+        strides (Tuple[int or Tuple[int]]): tuple of strides for each conv layer
+        n_res_layers (int): number of ``AttentionResidualBlocks`` to include
+        embedding_dim (int): size of hidden dimension of final output
+        **kwargs (dict): keyword arguments to be passed into ``SamePadConv3d`` and used by ``nn.Conv3d``
```

Comment:
hm do we really need kwargs vs making it explicit?

Reply:
kwargs are fed into `nn.Conv3d`. I'm not sure if there are any arguments for `nn.Conv3d` the user might want to have control of at this level. maybe `device`, which I could bring out explicitly, but then I should also bring that out in `SamePadConv3d`. `padding_mode` shouldn't be handled at this level. `dilation` I have not tested with the same padding. maybe I should just remove kwargs entirely. wdyt?